js/python-main.js

```diff
@@ -1745,6 +1650,68 @@ function web_editor(config) {
         });
     }
 
+    CONTROLLER.setup({
+        'setCode': function(code) {
+            EDITOR.setCode(code);
+        },
+        'getCode': function() {
+            return EDITOR.getCode();
+        },
+        'onCodeChange': function(callback) {
+            EDITOR.on_change(callback);
+        },
+        'loadHex': function(filename, hexStr) {
+            loadHex(filename, hexStr);
+        },
+        'loadFileToFs': function(filename, fileStr) {
+            loadFileToFilesystem(filename, fileStr);
+        },
+        'setMobileEditor': function(appFlash, appSave) {
+            // Show the Download and Flash buttons and remove Serial
+            $('#command-connect').hide();
+            $('#command-serial').hide();
+            $('#command-download').show();
+            $('#command-flash').show();
+            // Flash button to send the hex to the app
+            $('#command-flash').off('click', doFlash).on('click', function() {
+                try {
+                    updateMain();
+                    var output = FS.getUniversalHex();
+                } catch(e) {
+                    alert(config.translate.alerts.error + '\n' + e.message);
+                    return;
+                }
+                var filename = getSafeName() + '.hex';
+                appFlash(filename, output);
+            });
+            // Hex download buttons to send hex to the app
+            var mobileHexDownload = function() {
+                try {
+                    updateMain();
+                    var output = FS.getUniversalHex();
+                } catch(e) {
+                    alert(config.translate.alerts.error + '\n' + e.message);
+                    return;
+                }
+                var filename = getSafeName() + '.hex';
+                appSave(filename, output);
+            };
+            $('#command-download').off('click', doDownload).on('click', mobileHexDownload);
+            $('#save-hex').off('click', doDownload).on('click', mobileHexDownload);
+            doDownload = mobileHexDownload;
+            // Downloading a Python file now to send it to the app
+            downloadFileFromFilesystem = function(filename) {
+                // TODO: This was originally bytes
+                // var output = FS.readBytes(filename);
+                var output = FS.read(filename);
```

Comment:
@martinwork this was originally sending the data as a UInt8Array. Not sure how that gets serialise via postMessage, but sending a string matches the other messages a bit better.
Is this okay for the iOS app?

Reply:
I'll try it and see what arrives!

Reply:
Right now in his PR this is sending a normal string, the `UInt8Array` would be sent with the commented out line above this one.

Reply:
Okay, I'll remove the comment for merge, but let me know if you would need something other than a string.